main.nf

```diff
@@ -1436,6 +1466,62 @@ if (params.protein_searcher == 'sourmash'){
 }
 
 
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+/* --                                                                     -- */
+/* --             SEARCH NONCODING RNAS WITH INFERNAL                     -- */
+/* --                                                                     -- */
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+/*
+ * STEP 10 - Download/prepare Rfam databse
+ */
+if (params.search_noncoding && params.infernal_db) {
+  /*
+   * STEP 6 - unzip taxonomy information files for input to DIAMOND
+   */
+  if (hasExtension(params.infernal_db, "gz") ){
+    process gunzip_infernal_db {
+        tag "$gz"
+        publishDir path: { params.save_reference ? "${params.outdir}/infernal/database" : params.outdir },
+                   saveAs: { params.save_reference ? it : null }, mode: "${params.publish_dir_mode}"
+
+        input:
+        file gz from ch_infernal_db_gz
+
+        output:
+        file "${gz.baseName}" into ch_infernal_db
+
+        script:
+        """
+        gunzip -k --verbose --stdout --force ${gz} > ${gz.baseName}
+        """
+    }
+  }
+
+  process infernal_cmsearch {
+      tag "${sample_id}"
+      label "procss_high"
+      label "process_long"
+      publishDir "${params.outdir}/infernal/cmsearch", mode: "${params.publish_dir_mode}"
+
+      input:
+      file db from ch_infernal_db.collect()
+      set val(sample_id), file (fasta) from ch_noncoding_nucleotides
+
+      output:
+      file txt into ch_infernal_results
+
+      script:
+      txt = "${sample_id}.txt"
+      """
+      cmsearch --tblout ${txt} ${db} ${fasta}
```

Comment:
I was looking at the docstring there is --cpu flag and --mxsize flag that you can use to make this faster and the testing too probably http://manpages.ubuntu.com/manpages/xenial/man1/cmsearch.1.html